system_tests/app_engine/app/main.py

```diff
@@ -70,26 +77,38 @@ def capture():
         sys.stdout, sys.stderr = oldout, olderr
 
 
+def run_test_func(func):
+    with capture() as capsys:
+        try:
+            func()
+            return True, ''
+        except Exception as exc:
+            output = (
+                'Test {} failed: {}\n\n'
+                'Stacktrace:\n{}\n\n'
+                'Captured output:\n{}').format(
+                    func.func_name, exc, traceback.format_exc(),
+                    capsys.getvalue())
+            return False, output
+
+
 def run_tests():
     """Runs all tests.
 
     Returns:
         Tuple[bool, str]: A tuple containing True if all tests pass, False
         otherwise, and any captured output from the tests.
     """
-    status = False
+    status = True
     output = ''
 
-    with capture() as capsys:
-        try:
-            test_credentials()
-            test_default()
-            status = True
-        except Exception:
-            status = False
-            output = 'Stacktrace:\n{}\n'.format(traceback.format_exc())
-
-    output += 'Captured output:\n{}'.format(capsys.getvalue())
+    tests = (test_credentials, test_default)
+
+    for test in tests:
```

Comment:
If you do decide to make another commit for my nitpick, while you're in there you could just say:

```python
for test in (test_credentials, test_default):
    ...
```